### PR TITLE
LoginFilter added

### PR DESCRIPTION
**Pull request by SetMyDream**

### CODE REVIEW

File: src/main/java/com/internet/shop/dao/impl/UserDaoImpl.java [hunk 1]
```diff
@@ -40,4 +40,9 @@ public boolean delete(Long id) {
     public List<User> getAll() {
         return Storage.users;
     }
+
+    @Override
+    public Optional<User> findByLogin(String login) {
+        return Storage.users.stream().filter(s -> s.getLogin().equals(login)).findFirst();
```
Comment (KostiantynVorobiov) [inline src/main/java/com/internet/shop/dao/impl/UserDaoImpl.java]: Я не уверен но лучше с новой строки писать методы стримов

File: src/main/java/com/internet/shop/filter/AuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        HttpServletRequest req = (HttpServletRequest) request;
+        HttpServletResponse resp = (HttpServletResponse) response;
+        String url = req.getServletPath();
+        if (url.equals("/login") || url.equals("/registration")) {
+            filterChain.doFilter(req, resp);
+            return;
+        }
+        Long userId = (Long) req.getSession().getAttribute(USER_ID);
+        if (userId == null) {
+            resp.sendRedirect("/login");
```
Comment (KostiantynVorobiov) [inline src/main/java/com/internet/shop/filter/AuthenticationFilter.java]: resp.sendRedirect(request.getContextPath() + "/login");

**Review by KostiantynVorobiov — APPROVED** [review]
Исправ, и все навроде ок!

File: src/main/java/com/internet/shop/dao/impl/UserDaoImpl.java [hunk 3]
```diff
@@ -19,14 +19,19 @@ public User create(User user) {
     @Override
     public Optional<User> getById(Long id) {
         return getAll().stream()
-                .filter(user -> user.getId().equals(id))
+                .filter(user -> user
+                        .getId()
+                        .equals(id))
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/dao/impl/UserDaoImpl.java]:
```suggestion
                .filter(user -> user.getId().equals(id))
```

File: src/main/java/com/internet/shop/dao/impl/UserDaoImpl.java [hunk 4]
```diff
@@ -19,14 +19,19 @@ public User create(User user) {
...
 
     @Override
     public User update(User user) {
         IntStream.range(0, Storage.users.size())
-                .filter(i -> Storage.users.get(i).getId().equals(user.getId()))
+                .filter(i -> Storage.users
+                        .get(i)
+                        .getId()
+                        .equals(user.getId()))
                 .forEach(i -> Storage.users.set(i, user));
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/dao/impl/UserDaoImpl.java]: we writing each stream method from the new line, but not each method of lambda)

Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/dao/impl/UserDaoImpl.java]:
```suggestion
                .filter(i -> Storage.users.get(i).getId().equals(user.getId()))
                .forEach(i -> Storage.users.set(i, user));
```

File: src/main/java/com/internet/shop/dao/impl/UserDaoImpl.java [hunk 5]
```diff
@@ -40,4 +45,9 @@ public boolean delete(Long id) {
     public List<User> getAll() {
         return Storage.users;
     }
+
+    @Override
+    public Optional<User> findByLogin(String login) {
+        return Storage.users.stream().filter(s -> s.getLogin().equals(login)).findFirst();
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/dao/impl/UserDaoImpl.java]:
```suggestion
        return Storage.users.stream()
          .filter(s -> s.getLogin().equals(login))
          .findFirst();
```

File: src/main/java/com/internet/shop/filter/AuthenticationFilter.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        HttpServletResponse resp = (HttpServletResponse) response;
+        String url = req.getServletPath();
+        if (url.equals("/login") || url.equals("/registration")) {
+            filterChain.doFilter(req, resp);
+            return;
+        }
+        Long userId = (Long) req.getSession().getAttribute(USER_ID);
+        if (userId == null) {
+            resp.sendRedirect(req.getContextPath()
+                    + "/login");
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/filter/AuthenticationFilter.java]:
```suggestion
            resp.sendRedirect(req.getContextPath() + "/login");
```

File: src/main/java/com/internet/shop/filter/IpFilter.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class IpFilter implements Filter {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/filter/IpFilter.java]: this filter will only confuse you while testing other filters(cause your are testing only locally). It's good that you have figured how to do that, but let's remove it.

File: src/main/java/com/internet/shop/security/AuthenticationExceptionImpl.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
+package com.internet.shop.security;
+
+import com.internet.shop.exceptions.AuthenticationException;
+import com.internet.shop.lib.Inject;
+import com.internet.shop.lib.Service;
+import com.internet.shop.model.User;
+import com.internet.shop.service.UserService;
+
+@Service
+public class AuthenticationExceptionImpl implements AuthenticationService {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/security/AuthenticationExceptionImpl.java]:
```suggestion
public class AuthenticationServiceImpl implements AuthenticationService {
```

File: src/main/java/com/internet/shop/security/AuthenticationExceptionImpl.java [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String login, String password) throws AuthenticationException {
+        final String exceptionMessage = "Wrong login or password";
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/security/AuthenticationExceptionImpl.java]: let's make it constant

File: src/main/java/com/internet/shop/dao/impl/UserDaoImpl.java [hunk 10]
```diff
@@ -40,4 +40,11 @@ public boolean delete(Long id) {
     public List<User> getAll() {
         return Storage.users;
     }
+
+    @Override
+    public Optional<User> findByLogin(String login) {
+        return Storage.users.stream()
+                .filter(s -> s.getLogin().equals(login))
+                .findFirst();
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/dao/impl/UserDaoImpl.java]:
```suggestion
                .filter(u -> u.getLogin().equals(login))
                .findFirst();
```

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**